code/game/objects/items/weapons/grenades/grenade.dm

```diff
@@ -64,14 +64,15 @@
 		det_time *= RAND_DECIMAL(1-variance, 1+variance)
 
 	spawn(det_time)
-		prime()
+		prime(user)
 		return
```

Comment:
use addtimer

Reply:
Why though?

Reply:
Testing revealed issues, primarily undefined vars and conflicts in other areas. Works as it, might refactor later when I have more time.